sdk/keyvault/azure-keyvault-keys/tests/test_crypto_client.py

```diff
@@ -289,23 +289,57 @@ def test_operations(key, expected_error_substrings, encrypt_algorithms, wrap_alg
         the_year_3000 = datetime(3000, 1, 1, tzinfo=_UTC)
 
         rsa_wrap_algorithms = [algo for algo in KeyWrapAlgorithm if algo.startswith("RSA")]
+        rsa_encryption_algorithms = [algo for algo in EncryptionAlgorithm if algo.startswith("RSA")]
         key_name = self.get_resource_name("rsa-not-yet-valid")
         not_yet_valid_key = key_client.create_rsa_key(key_name, not_before=the_year_3000)
-        test_operations(not_yet_valid_key, [str(the_year_3000)], EncryptionAlgorithm, rsa_wrap_algorithms)
+        test_operations(not_yet_valid_key, [str(the_year_3000)], rsa_encryption_algorithms, rsa_wrap_algorithms)
 
         # nor should they succeed with a key whose exp has passed
         the_year_2000 = datetime(2000, 1, 1, tzinfo=_UTC)
 
         key_name = self.get_resource_name("rsa-expired")
         expired_key = key_client.create_rsa_key(key_name, expires_on=the_year_2000)
-        test_operations(expired_key, [str(the_year_2000)], EncryptionAlgorithm, rsa_wrap_algorithms)
+        test_operations(expired_key, [str(the_year_2000)], rsa_encryption_algorithms, rsa_wrap_algorithms)
 
         # when exp and nbf are set, error messages should contain both
         the_year_3001 = datetime(3001, 1, 1, tzinfo=_UTC)
 
         key_name = self.get_resource_name("rsa-valid")
         valid_key = key_client.create_rsa_key(key_name, not_before=the_year_3000, expires_on=the_year_3001)
-        test_operations(valid_key, (str(the_year_3000), str(the_year_3001)), EncryptionAlgorithm, rsa_wrap_algorithms)
+        test_operations(
+            valid_key, (str(the_year_3000), str(the_year_3001)), rsa_encryption_algorithms, rsa_wrap_algorithms
+        )
+
+    @KeyVaultPreparer()
+    def test_encrypt_valid_arguments(self, azure_keyvault_url, **kwargs):
```

Comment:
Because they don't need to do any crypto, you could make these tests more efficient by using a mock key instead (`test_prefers_local_provider` is an example of what that looks like).